api/src/view_function.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    accept_type::AcceptType,
+    failpoint::fail_point_poem,
+    response::{
+        BadRequestError, BasicErrorWith404, BasicResponse, BasicResponseStatus, BasicResultWith404,
+    },
+    ApiTags, Context,
+};
+use anyhow::Context as AnyhowContext;
+use aptos_api_types::{
+    verify_function_identifier, verify_module_identifier, Address, AptosErrorCode,
+    IdentifierWrapper, MoveValue, ViewRequest, U64,
+};
+use aptos_vm::AptosVM;
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{ModuleId, TypeTag},
+};
+use poem_openapi::{
+    param::{Path, Query},
+    payload::Json,
+    OpenApi,
+};
+use std::{convert::TryInto, sync::Arc};
+
+/// API for executing Move view function.
+pub struct ViewFunctionApi {
+    pub context: Arc<Context>,
+}
+
+#[OpenApi]
+impl ViewFunctionApi {
+    /// Get account resource
+    ///
+    /// Retrieves an individual resource from a given account and at a specific ledger version. If the
+    /// ledger version is not specified in the request, the latest ledger version is used.
+    ///
+    /// The Aptos nodes prune account state history, via a configurable time window.
+    /// If the requested ledger version has been pruned, the server responds with a 410.
+    #[oai(
+        path = "/view/:address/module/:module_name/function/:function_name",
+        method = "post",
+        operation_id = "view",
+        tag = "ApiTags::View"
+    )]
+    async fn view_function(
+        &self,
+        accept_type: AcceptType,
+        /// Address of account with or without a `0x` prefi
+        address: Path<Address>,
+        /// Name of module to execute e.g. `coin`
+        module_name: Path<IdentifierWrapper>,
+        /// Name of function to execute e.g. `coin`
+        function_name: Path<IdentifierWrapper>,
+        /// Type and vale arguments to the function,
+        request: Json<ViewRequest>,
+        /// Ledger version to get state of account
+        ///
+        /// If not provided, it will be the latest version
+        ledger_version: Query<Option<U64>>,
+    ) -> BasicResultWith404<Vec<MoveValue>> {
```

Comment:
The return `MoveValue` is actually quite nasty...It was converted to an untyped string and IMHO won't serve well as the return type.

Reply:
This is only used for "JSON", which already loses all of its typing anyways.  BCS provides the true typing right now.